# Project: Students giving up courses

## Problem definition
In a educational platform, a lot of students are giving up their courses before finishing, causing some losses to the platform. For that reason, it's essential to analyse students' academic behaviour, which lets us know factors that might be associated with leaving.

## Objective
- Analyse academic variables for the students
- Build a model that predict the probability of leaving a course 

## Research questions

1. What proportion of students abandon their courses?
2. Which variables are associated with course abandonment?
3. Do students who study fewer hours leave courese more frequently?
4. Could course abandonment be predicted abandonment using machine learning?

## Importing libraries

In [1]:
import pandas as pd

## Reading dataset

In [2]:
df = pd.read_csv('datasets/xAPI-Edu-Data.csv')

df.head()

,gender,NationalITy,PlaceofBirth,StageID,GradeID,SectionID,Topic,Semester,Relation,raisedhands,VisITedResources,AnnouncementsView,Discussion,ParentAnsweringSurvey,ParentschoolSatisfaction,StudentAbsenceDays,Class
0,M,KW,KuwaIT,lowerlevel,G-04,A,IT,F,Father,15,16,2,20,Yes,Good,Under-7,M
1,M,KW,KuwaIT,lowerlevel,G-04,A,IT,F,Father,20,20,3,25,Yes,Good,Under-7,M
2,M,KW,KuwaIT,lowerlevel,G-04,A,IT,F,Father,10,7,0,30,No,Bad,Above-7,L
3,M,KW,KuwaIT,lowerlevel,G-04,A,IT,F,Father,30,25,5,35,No,Bad,Above-7,L
4,M,KW,KuwaIT,lowerlevel,G-04,A,IT,F,Father,40,50,12,50,No,Bad,Above-7,M


## Exploratory analysis

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 480 entries, 0 to 479
Data columns (total 17 columns):
 #   Column                    Non-Null Count  Dtype 
---  ------                    --------------  ----- 
 0   gender                    480 non-null    object
 1   NationalITy               480 non-null    object
 2   PlaceofBirth              480 non-null    object
 3   StageID                   480 non-null    object
 4   GradeID                   480 non-null    object
 5   SectionID                 480 non-null    object
 6   Topic                     480 non-null    object
 7   Semester                  480 non-null    object
 8   Relation                  480 non-null    object
 9   raisedhands               480 non-null    int64 
 10  VisITedResources          480 non-null    int64 
 11  AnnouncementsView         480 non-null    int64 
 12  Discussion                480 non-null    int64 
 13  ParentAnsweringSurvey     480 non-null    object
 14  ParentschoolSatisfaction  

According to the output of `df.info()`, the dataset contains 480 records and 17 variables, including 4 numerical variables and 13 categorical (object) variables. The meaning of each variable is described below:

- gender: student's sex